cli/src/commands/init.rs

```diff
@@ -163,3 +113,58 @@ fn print_trackable_remote_branches(ui: &Ui, view: &View) -> io::Result<()> {
     )?;
     Ok(())
 }
+
+// TODO(essiene): Move to cli/src/commands/git.rs for `jj git init`
+fn git_init(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    workspace_root: &Path,
+    use_git_backend: bool,
+    git_repo: &Option<String>,
+) -> Result<(), CommandError> {
+    let cwd = command.cwd().canonicalize().unwrap();
+    let relative_wc_path = file_util::relative_path(&cwd, workspace_root);
+
+    if git_repo.is_none() {
+        if use_git_backend {
+            if workspace_root.join(".git").exists() {
+                return Err(user_error_with_hint(
+                    "Did not create a jj repo because there is an existing Git repo in this \
+                     directory.",
+                    format!(
+                        r#"To create a repo backed by the existing Git repo, run `jj init --git-repo={}` instead."#,
+                        relative_wc_path.display()
+                    ),
+                ));
+            }
+
+            Workspace::init_internal_git(command.settings(), workspace_root)?;
+        }
+    } else if let Some(git_store_str) = &git_repo {
```

Comment:
nit: maybe better to test this first because `git_repo.is_none()` can be just `} else {`

Reply:
Whoops! thanks. I missed that case before. I've now changed this to a match statement, so it's impossible to miss ignored cases.